### PR TITLE
Create PubSubIO Load test

### PR DESCRIPTION
**Pull request by akashorabek**
This PR adds Load tests for PubSubIO. There are several configurations depending on the amount of data and the runner where the tests will be executed:

- **local**: Data volume is 0.2 MB. The test runs on the local machine using the Direct Runner.
- **medium**: Data volume is 10 GB. The test runs on the Dataflow runner.
- **large**: Data volume is 100 GB. The test runs on the Dataflow runner.

How to verify the results:

1. Install Gradle on your local machine.
2. Run the command `gradle :it:google-cloud-platform:PubSubPerformanceTest --tests "org.apache.beam.it.gcp.pubsub.PubSubIOLT" -d` from the project root to execute the tests.
3. The tests will pass successfully if all data is processed within a certain amount of time and the processed data matches the expected amount.
4. Additionally, upon running the test, a link for monitoring the job will be displayed in the terminal, providing more detailed information on how much data has been processed and the progress. An example link is: https://console.cloud.google.com/dataflow/jobs/us-central1/2024-02-13_00_10_16-5567586527775732420?project=apache-beam-testing.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by akashorabek** [comment]
R: @damccorm @Abacn

File: it/google-cloud-platform/build.gradle [hunk 1]
```diff
@@ -72,13 +73,16 @@ dependencies {
 
     testImplementation project(path: ":sdks:java:testing:test-utils")
     testImplementation project(path: ":sdks:java:io:google-cloud-platform")
+    testImplementation project(path: ":sdks:java:extensions:protobuf", configuration: "testRuntimeMigration")
     testImplementation project(path: ":sdks:java:io:synthetic")
     testImplementation library.java.mockito_inline
     testImplementation project(path: ":sdks:java:extensions:google-cloud-platform-core", configuration: "testRuntimeMigration")
+    testImplementation project(path: ":runners:direct-java", configuration: "shadow")
```
Comment (Abacn) [inline it/google-cloud-platform/build.gradle:80]: Why this is needed. There are "testRuntimeOnly" below.

If it is the case, then we can remove the line below.

**Review by Abacn — COMMENTED** [review]
Thanks. Had a couple of initial comments. Is there a job id available for demonstration?

File: it/google-cloud-platform/build.gradle (hunk 1 above)
Comment (akashorabek) [reply it/google-cloud-platform/build.gradle:80]: Removed the "testRuntimeOnly" line. We are using some classes from this dependency and IDE couldn't locate those without "testImplementation".

File: it/google-cloud-platform/build.gradle [hunk 2]
```diff
@@ -41,6 +41,7 @@ dependencies {
     implementation 'org.apache.parquet:parquet-avro:1.12.0'
     implementation 'org.apache.parquet:parquet-common:1.12.0'
     implementation 'org.apache.parquet:parquet-hadoop:1.12.0'
+    implementation 'org.apache.commons:commons-lang3:3.9@jar'
```
[the remark this answers is not in the document]
Comment (akashorabek) [reply it/google-cloud-platform/build.gradle]: Removed the `@jar` suffix.

**Comment by akashorabek** [comment]
> Thanks. Had a couple of initial comments. Is there a job id available for demonstration?

Here is the links to one of the test jobs:
Write job: https://console.cloud.google.com/dataflow/jobs/us-central1/2024-02-21_02_10_37-12171205265624652554;step=Read%20from%20source;mainTab=JOB_GRAPH;bottomTab=JOB_ERROR_REPORTING;bottomStepTab=DATA_SAMPLING;logsSeverity=INFO;graphView=0?project=apache-beam-testing&pageState=(%22dfTime%22:(%22l%22:%22dfJobMaxTime%22))
Read job: https://console.cloud.google.com/dataflow/jobs/us-central1/2024-02-21_02_11_03-1874843895021090331;step=Read%20from%20PubSub;mainTab=JOB_GRAPH;bottomTab=DATA_SAMPLING;logsSeverity=INFO;graphView=0?project=apache-beam-testing

File: it/google-cloud-platform/src/test/java/org/apache/beam/it/gcp/pubsub/PubSubIOLT.java [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+            String.format(
+                "Unknown test configuration: [%s]. Pass to a valid configuration json, or use"
+                    + " config presets: %s",
+                testConfigName, TEST_CONFIGS_PRESET.keySet()));
+      }
+    }
+
+    // Explicitly set up number of bundles in SyntheticUnboundedSource since it has a bug in
+    // implementation where
+    // number of lost data in streaming pipeline equals to number of initial bundles.
```
Comment (Abacn) [inline it/google-cloud-platform/src/test/java/org/apache/beam/it/gcp/pubsub/PubSubIOLT.java:142]: I see `2024-02-21_02_10_37-12171205265624652554` writes `99,999,980` record. Why there are "lost data in streaming pipeline"? Is this a bug in SDK?

Comment (Abacn) [reply it/google-cloud-platform/src/test/java/org/apache/beam/it/gcp/pubsub/PubSubIOLT.java:142]: Thanks for the explanation. Synthetic source is for testing so this is less concern.LGTM

**Review by Abacn — APPROVED** [review]
Thanks. For SyntheticUnboundedSource issue and the Kafka test comment good to open a GitHub issue to track it.

**Pull request merged**